packages/adblocker-content/adblocker.ts

```diff
@@ -31,6 +31,18 @@ export interface IMessageFromBackground {
   }[];
 }
 
+function debounce<F extends (...args: any[]) => any>(fn: F, waitFor: number) {
+  let timeout: NodeJS.Timeout | undefined;
+
+  return (...args: Parameters<F>) => {
+    clearTimeout(timeout);
```

Comment:
```suggestion
function debounce<F extends (...args: any[]) => any>(fn: F, waitFor: number, maxWaitFor: number) {
  let timeout: NodeJS.Timeout | undefined;
  let firstCallAt: number | undefined;

  return (...args: Parameters<F>) => {
    if (!firstCallAt) {
      firstCallAt = Date.now();
    }

    clearTimeout(timeout);

    if (Date.now() - firstCallAt >= maxWaitFor) {
      try {
        fn(...args);
      } finally {
        firstCallAt = undefined;
      }
      return;
    }
```

Reply:
Thanks for the idea of time-based thresholding. This is being handled in https://github.com/seia-soto/adblocker/pull/7